pkgs/applications/science/astronomy/galario/default.nix

```
@@ -0,0 +1,75 @@
+{ stdenv
+, fetchurl
+, cmake
+, fftw
+, fftwFloat
+, enablePython ? false
+, pythonPackages
+, llvmPackages
+}:
+
+stdenv.mkDerivation rec {
+  name = "galario-${version}";
+  version = "1.2.1";
+
+  srcs = [
+    (fetchurl {
+      url = "https://github.com/mtazzari/galario/archive/v${version}.tar.gz";
+      sha256 = "1m1lc6z5db6h2l719080jlh2aaaa82p1ddfzsv0kqlh6biyx6psq";
+    })
+    (fetchurl {
+      url = "https://github.com/UCL/GreatCMakeCookOff/archive/v2.1.9.tar.gz";
+      sha256 = "0b31q59x8kjhnnkdqryafkag55nd2y86lj4wac0wz58aigi24fz3";
+    })
+  ];
+
+  nativeBuildInputs = [ cmake ];
+
+  buildInputs = [ fftw fftwFloat ]
+    ++ stdenv.lib.optional enablePython pythonPackages.python
+    ++ stdenv.lib.optional stdenv.isDarwin llvmPackages.openmp
+  ;
+
+  propagatedBuildInputs = stdenv.lib.optional enablePython [
+    pythonPackages.numpy
+    pythonPackages.cython
+    pythonPackages.pytest
+  ];
+
+  checkInputs = stdenv.lib.optional enablePython pythonPackages.scipy;
+
+  sourceRoot = "${name}";
+
+  preConfigure = ''
+    mkdir -p build/external/src/
+    cp -r ../GreatCMakeCookOff-2.1.9 build/external/src/GreatCMakeCookOff
```

If you just copy that single file, I'd only add the regular project as `src` and have this in a let binding and then do this:

```suggestion
    cp -r ${greatCmakeCookOff} build/external/src/GreatCMakeCookOff
```

---

I'm not sure to understand your suggestion. Can you please elaborate a bit?

---

```diff
diff --git i/pkgs/applications/science/astronomy/galario/default.nix w/pkgs/applications/science/astronomy/galario/default.nix
index e3b45bc0944..08450e56983 100644
--- i/pkgs/applications/science/astronomy/galario/default.nix
+++ w/pkgs/applications/science/astronomy/galario/default.nix
@@ -1,5 +1,6 @@
 { stdenv
-, fetchurl
+, fetchzip
+, fetchFromGitHub
 , cmake
 , fftw
 , fftwFloat
@@ -7,22 +8,24 @@
 , pythonPackages
 , llvmPackages
 }:
-
+let
+  # CMake recipes are needed to build galario
+  # Build process would usually download them
+  great-cmake-cookoff = fetchzip {
+    url = "https://github.com/UCL/GreatCMakeCookOff/archive/v2.1.9.tar.gz";
+    sha256 = "1yd53b5gx38g6f44jmjk4lc4igs3p25z6616hfb7aq79ly01q0w2";
+  };
+in
 stdenv.mkDerivation rec {
   pname = "galario";
   version = "1.2.1";
 
-  srcs = [
-    (fetchurl {
-      url = "https://github.com/mtazzari/galario/archive/v${version}.tar.gz";
-      sha256 = "1m1lc6z5db6h2l719080jlh2aaaa82p1ddfzsv0kqlh6biyx6psq";
-    })
-    (fetchurl {
-      # These cmake recipes are needed to build galario
-      url = "https://github.com/UCL/GreatCMakeCookOff/archive/v2.1.9.tar.gz";
-      sha256 = "0b31q59x8kjhnnkdqryafkag55nd2y86lj4wac0wz58aigi24fz3";
-    })
-  ];
+  src = fetchFromGitHub {
+    owner = "mtazzari";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "1akz7md7ly16a89zr880c265habakqdg9sj8iil90klqa0i21w6g";
+  };
 
   nativeBuildInputs = [ cmake ];
 
@@ -39,11 +42,10 @@ stdenv.mkDerivation rec {
 
   checkInputs = stdenv.lib.optional enablePython pythonPackages.scipy;
 
-  sourceRoot = "${pname}-${version}";
-
   preConfigure = ''
-    mkdir -p build/external/src/
-    cp -r ../GreatCMakeCookOff-2.1.9 build/external/src/GreatCMakeCookOff
+    mkdir -p build/external/src
+    cp -r ${great-cmake-cookoff} build/external/src/GreatCMakeCookOff
+    chmod -R 777 build/external/src/GreatCMakeCookOff
   '';
 
   preCheck = ''
```

---

Feel free to apply that patch. Or ask for explanation :)

---

I've applied the patch. Thanks for the suggestions.